src/data-pipeline/spark-etl/src/main/java/software/aws/solution/clickstream/transformer/BaseTransformerV3.java

```diff
@@ -84,6 +85,7 @@ public Dataset<Row> extractEvent(final Dataset<Row> convertedDataset) {
         List<String> allFields = ModelV2.getEventFields();
         Dataset<Row> eventDataset = convertedDataset.select(explode(expr("dataOut.events")).alias("event"))
                 .select("event.*")
+                .withColumn(Constant.TRAFFIC_SOURCE_SOURCE, coalesce(col(Constant.TRAFFIC_SOURCE_SOURCE), lit(DIRECT)))
```

Comment:
any requirement for assigning it to DIRECT?

we could treat null as unknown while disabling traffic source parsing, the value DIRECT has different meanings.